### PR TITLE
Update alert processig and filtering

### PR DESCRIPTION
**Pull request by pbalogh-sa**
Signed-off-by: Peter Balogh <p.balogh.sa@gmail.com>

| Q               | A
| --------------- | ---
| Bug fix?        |yes
| New feature?    | no
| API breaks?     | no
| Deprecations?   | no
| Related tickets | fixes #X, partially #Y, mentioned in #Z
| License         | Apache 2.0

### What's in this PR?
Update alert filtering and processing

### CODE REVIEW

File: internal/alertmanager/dispatcher/dispatcher.go [hunk 1]
```diff
@@ -52,7 +52,7 @@ func alertFilter(promAlerts []model.Alert) []model.Alert {
 
 	filteredAlerts := []model.Alert{}
 	for _, alert := range promAlerts {
-		if _, labelOK := alert.Labels["kafka_cr"]; labelOK {
+		if label, labelOK := alert.Labels["alertGroup"]; labelOK && label == "kafka" {
```
Comment (stoader) [inline internal/alertmanager/dispatcher/dispatcher.go]: This doesn't look to be backward compatible with  previous deployments. I'd do the filtering based on command.

File: config/samples/kafkacluster-prometheus.yaml [hunk 2]
```diff
@@ -115,6 +118,7 @@ spec:
       for: 3m
       labels:
         severity: alert
+        alretGroup: kafka
```
Comment (stoader) [inline config/samples/kafkacluster-prometheus.yaml]: typo

File: config/samples/kafkacluster-prometheus.yaml [hunk 3]
```diff
@@ -89,6 +90,7 @@ spec:
       for: 5m
       labels:
         severity: alert
+        alretGroup: kafka
```
Comment (stoader) [inline config/samples/kafkacluster-prometheus.yaml]: typo

Comment (baluchicken) [inline config/samples/kafkacluster-prometheus.yaml]:
```suggestion
        alertGroup: kafka
```

File: config/samples/kafkacluster-prometheus.yaml (hunk 2 above)
Comment (baluchicken) [inline config/samples/kafkacluster-prometheus.yaml]:
```suggestion
        alertGroup: kafka
```

File: internal/alertmanager/currentalert/process.go [hunk 4]
```diff
@@ -91,11 +94,17 @@ func (e *examiner) processAlert(ds disableScaling) (bool, error) {
 
 	switch e.Alert.Annotations["command"] {
 	case "addPVC":
+		if err := checkLabelExists(e.Alert.Labels, "persistentvolumeclaim"); err != nil {
```
Comment (stoader) [inline internal/alertmanager/currentalert/process.go]: I'd introduce types for alerts and define a validation function which could be invoked after filtering.

File: internal/alertmanager/currentalert/process.go [hunk 5]
```diff
@@ -91,11 +94,17 @@ func (e *examiner) processAlert(ds disableScaling) (bool, error) {
...
+			return false, emperror.Wrap(err, "label: persistentvolumeclaim")
+		}
 		err := addPVC(e.Alert.Labels, e.Alert.Annotations, e.Client)
 		if err != nil {
 			return false, err
 		}
 	case "downScale":
+		if err := checkLabelExists(e.Alert.Labels, "kafka_cr"); err != nil {
```
Comment (stoader) [inline internal/alertmanager/currentalert/process.go]: I'd introduce types for alerts and define a validation function which could be invoked after filtering.

File: internal/alertmanager/currentalert/process.go [hunk 6]
```diff
@@ -107,6 +116,9 @@ func (e *examiner) processAlert(ds disableScaling) (bool, error) {
 
 		return true, nil
 	case "upScale":
+		if err := checkLabelExists(e.Alert.Labels, "kafka_cr"); err != nil {
```
Comment (stoader) [inline internal/alertmanager/currentalert/process.go]: I'd introduce types for alerts and define a validation function which could be invoked after filtering.

File: config/samples/kafkacluster-prometheus.yaml [hunk 7]
```diff
@@ -114,7 +117,8 @@ spec:
       expr: min(kafka_server_replicamanager_partitioncount)  by (namespace, kafka_cr) < 40
       for: 3m
       labels:
-        severity: alert
+        severity: alerta
```
Comment (amuraru) [inline config/samples/kafkacluster-prometheus.yaml]: Typo?

Comment (pbalogh-sa) [reply config/samples/kafkacluster-prometheus.yaml]: thx

**Pull request merged**